web-common/src/layout/app-store.ts

```diff
@@ -14,54 +17,91 @@ export interface ActiveEntity {
  * App wide store to store metadata
  * Currently caches active entity from URL
  */
-export interface AppStore {
+interface AppStore {
   activeEntity: ActiveEntity;
   previousActiveEntity: ActiveEntity;
 }
 
-// We should rewrite ActiveEntity using appScreen dervied store
-const { update, subscribe } = writable({
+export const appScreen = derived(page, ($page) => {
+  let activeEntity;
+  switch ($page.route.id) {
+    case "/(application)":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Home,
+      };
+      break;
+    case "/(application)/source/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Source,
+      };
+      break;
+    case "/(application)/model/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Model,
+      };
+      break;
+    case "/(application)/dashboard/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Dashboard,
+      };
+      break;
+    case "/(application)/dashboard/[name]/edit":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.MetricsDefinition,
+      };
+      break;
+    case "/(application)/welcome":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Splash,
+      };
+      break;
+    case "/[organization]/[project]/[dashboard]":
+      activeEntity = {
+        name: $page?.params?.dashboard,
+        type: MetricsEventScreenName.Dashboard,
+      };
+      break;
+    default:
+      // Return home as default
+      activeEntity = { name: "", type: MetricsEventScreenName.Home };
+  }
+
+  appStore.setActiveEntity(
+    activeEntity.name,
+    ScreenToEntityMap[activeEntity.type]
+  );
+  return activeEntity;
+});
+
+// App store is being utilized for making previous entity inactive in the HTTP request queue
+const { update } = writable({
   activeEntity: undefined,
   previousActiveEntity: undefined,
 } as AppStore);
 
 const appStoreReducers = {
   setActiveEntity(name: string, type: EntityType) {
     update((state) => {
+      state.previousActiveEntity = state.activeEntity;
```

Comment:
Why is this moved to the top? Wouldnt this make the current entity inactive? Either move this to the bottom or move the call to `setActiveEntity` to the top in `appScreen` body.

Reply:
The previous implementation had a bug. Corrected it with these changes. You can verify by console logging `state.previousActiveEntity.name` when it's being made inactive. 

Now when you move from asset A to asset B, asset A is being made inactive now.

`name` provided through props is the current active entity being set. `state.activeEntity` just before being updated is now the previous active entity which has to be made inactive.